test/int/integration_suite_test.go

```diff
@@ -110,7 +110,10 @@ var _ = BeforeSuite(func(done Done) {
 		Expect(err).ToNot(HaveOccurred())
 	}()
 
-	k8sClient = k8sManager.GetClient()
+	// It's recommended to construct the client directly for tests
+	// see https://github.com/kubernetes-sigs/controller-runtime/issues/343#issuecomment-469435686
+	k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
```

Comment:
this is the main treatment against caches - the client from `k8sManager` establishes caches (and shared informers?) for GET requests so for tests it's necessary to build the simple direct client